test/lib/jdk/test/lib/process/ProcessTools.java

```diff
@@ -217,14 +217,20 @@ protected void processLine(String line) {
         try {
             if (timeout > -1) {
                 if (timeout == 0) {
-                    latch.await();
+                    while (!latch.await(1, TimeUnit.SECONDS)) {
+                        if (!p.isAlive()) {
+                            //Fail if process finished before printed expected string
+                            latch.countDown();
+                            throw new RuntimeException("Started process is not alive.");
```

Comment:
Please add the name to the exception; it will be easier to determine what failed.